IdentityCore/src/intune/MSIDIntuneEnrollmentIdsCache.m

```diff
@@ -140,27 +140,32 @@ - (NSString *)enrollmentIdForHomeAccountId:(NSString *)homeAccountId
 }
 
 - (NSString *)enrollmentIdForHomeAccountId:(NSString *)homeAccountId
-                                    legacyUserId:(NSString *)legacyUserId
+                              legacyUserId:(NSString *)legacyUserId
                                    context:(id<MSIDRequestContext>)context
                                      error:(NSError **)error
 {
+    NSString *enrollmentId = nil;
+    
+    // If homeAccountID is provided, try to match by it first.
     if (homeAccountId)
     {
-        // If homeAccountID is provided, always require an exact match
-        return [self enrollmentIdForUserId:homeAccountId context:context error:error];
+        enrollmentId = [self enrollmentIdForHomeAccountId:homeAccountId context:context error:error];
     }
-    else
+    if (enrollmentId)
     {
-        // If legacy userID is provided and we didn't find an exact match, do a fallback to any enrollment ID to support no userID or single userID scenarios
-        NSString *enrollmentID = legacyUserId ? [self enrollmentIdForUserId:legacyUserId context:context error:error] : nil;
-        if (enrollmentID)
-        {
-            return enrollmentID;
-        }
-        
-        enrollmentID = [self enrollmentIdIfAvailableWithContext:context error:error];
-        return enrollmentID;
+        return enrollmentId;
     }
+    
+    // If legacy userID is provided, try to match by userID.
+    enrollmentId = legacyUserId ? [self enrollmentIdForUserId:legacyUserId context:context error:error] : nil;
+    if (enrollmentId)
+    {
+        return enrollmentId;
+    }
+    
```

Comment:
nit: we are checking too many things above.
why not just do if (legacyUserId) { enrollmentId = ... ;  return this if not nil}
followed by returning [self enrollmendIdAva...]

Reply:
You're right :) I just edited the code we had in this place before.